src/features/student/views/StudentTraceView/components/UpdateTraceForm/UpdateTraceForm.vue

```diff
@@ -0,0 +1,173 @@
+<script setup lang="ts">
+import type { TraceDetailDTO } from '@/api/avenir-esr'
+import { useDateUtils } from '@/common/composables'
+import TraceAiJustificationTextareaFormField from '@/features/student/views/StudentToolsTracesView/components/TraceAiJustificationTextareaFormField/TraceAiJustificationTextareaFormField.vue'
+import TraceAiUsageToggleFormField from '@/features/student/views/StudentToolsTracesView/components/TraceAiUsageToggleFormField/TraceAiUsageToggleFormField.vue'
+import TraceAuthenticDeclarationToggleFormField from '@/features/student/views/StudentToolsTracesView/components/TraceAuthenticDeclarationToggleFormField/TraceAuthenticDeclarationToggleFormField.vue'
+import TraceFileUploadFormField from '@/features/student/views/StudentToolsTracesView/components/TraceFileUploadFormField/TraceFileUploadFormField.vue'
+import TraceGroupProductionToggleFormField from '@/features/student/views/StudentToolsTracesView/components/TraceGroupProductionToggleFormField/TraceGroupProductionToggleFormField.vue'
+import TraceNameInputFormField from '@/features/student/views/StudentToolsTracesView/components/TraceNameInputFormField/TraceNameInputFormField.vue'
+import TracePersonalNoteTextareaFormField from '@/features/student/views/StudentToolsTracesView/components/TracePersonalNoteTextareaFormField/TracePersonalNoteTextareaFormField.vue'
+import { useUpdateTraceForm } from '@/features/student/views/StudentTraceView/components/UpdateTraceForm/use-update-trace-form/use-update-trace-form'
+import { useToasterStore, useTracesStore } from '@/store'
+import { useI18n } from 'vue-i18n'
+
+const { trace } = defineProps<UpdateTraceFormProps>()
+
+const { addSuccessMessage, } = useToasterStore()
+const { hideUpdateTraceModal } = useTracesStore()
+
+interface UpdateTraceFormProps {
+  trace: TraceDetailDTO
+}
+
+const { t } = useI18n()
+const { setUpdateTraceForm, setUpdateTraceFormValid } = useTracesStore()
+
+function onTraceUpdated () {
+  addSuccessMessage({
+    timeout: 2000,
+    description: t('student.views.studentTraceView.updateTraceModal.success')
+  })
+  hideUpdateTraceModal()
+}
+
+const { form, isFormValid } = useUpdateTraceForm(trace, onTraceUpdated)
+const { formatTranslatedDateTime } = useDateUtils()
+
+const useIAField = form.useField({ name: 'useIA' })
+const iaJustificationField = form.useField({ name: 'iaJustification' })
+
+const showIAJustification = computed(() => useIAField.state.value.value === true)
+
+function onChangeAiUsageToggle (value: boolean) {
+  if (!value) {
+    iaJustificationField.api.handleChange('')
+  }
+}
+const traceFileUploadLabel = computed<string>(() => {
```

Comment:
```suggestion
}

const traceFileUploadLabel = computed(() => {
```
Aération du code + typage explicite non nécessaire